shadows/framework/src/main/java/org/robolectric/shadows/ShadowNetworkCapabilities.java

```diff
@@ -54,6 +57,26 @@ public NetworkCapabilities removeCapability(int capability) {
         .removeCapability(capability);
   }
 
+  /**
+   * Changes {@link NetworkSpecifier} for this network capabilities. Works only on Android O and
+   * higher. For lower versions use {@link #setNetworkSpecifier(String)}
+   */
+  @Implementation(minSdk = O)
+  public NetworkCapabilities setNetworkSpecifier(NetworkSpecifier networkSpecifier) {
+    return reflector(NetworkCapabilitiesReflector.class, realNetworkCapabilities)
+        .setNetworkSpecifier(networkSpecifier);
+  }
+
+  /**
+   * Changes {@link NetworkSpecifier} for this network capabilities. Works only on Android N_MR1 and
+   * lower. For higher versions use {@link #setNetworkSpecifier(NetworkSpecifier)}
+   */
+  @Implementation(maxSdk = N_MR1)
```

Comment:
This should be `@Implementation(minSdk = N, maxSdk = N_MR1)`

Looks like from L (SDK 21) to M (SDK 23), the signature of this method was

`public void setNetworkSpecifier(String networkSpecifier) {...}`:

http://androidxref.com/5.0.0_r2/xref/frameworks/base/core/java/android/net/NetworkCapabilities.java#461

Strangely that does not result in a compile error here on Robolectric's GitHub CI, but we can look into that separately.

Reply:
We can use `looseSignature` and  SDK version checking in implementation method to support it. IMO, it's enough to support `setNetworkSpecifier` from `L` in this PR. 

> Strangely that does not result in a compile error here on Robolectric's GitHub CI, but we can look into that separately.

Maybe it can be a part of GSoC 2022's `shadowOf` improvementation.

Reply:
Do I need to just add `looseSignatures = true` to `@Implements` annotation on `ShadowNetworkCapabilities` class? Or something else?

Reply:
@hoisie what do you think? IMO, this PR is fine to merge without using looseSignature to add older SDKs.